### PR TITLE
Add EOS EVM

### PR DESCRIPTION
**Pull request by DenisCarriere**
## Explanation

Suggest adding EOS EVM to Metamask

## Chains
- `17777` EOS EVM Network
https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-17777.json
- `15557` EOS EVM Network Testnet
https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-15557.json

### EOS EVM References
- https://eosnetwork.com/eos-evm
- https://docs.eosnetwork.com/docs/latest/eos-evm/

### CODE REVIEW

**Comment by DenisCarriere** [comment]
I have read the CLA Document and I hereby sign the CLA

File: shared/constants/network.ts [hunk 1]
```diff
@@ -177,6 +179,7 @@ export const HARMONY_DISPLAY_NAME = 'Harmony Mainnet Shard 0';
 export const PALM_DISPLAY_NAME = 'Palm';
 export const AURORA_DISPLAY_NAME = 'Aurora Mainnet';
 export const CELO_DISPLAY_NAME = 'Celo Mainnet';
+export const EOS_DISPLAY_NAME = 'EOS EVM Network';
```
Comment (kj4ezj) [inline shared/constants/network.ts:183]: **Optional**  
I would drop "Network" here. Our branding just calls it EOS EVM, no "network" suffix.
https://eosnetwork.com/eos-evm

**Review by kj4ezj — DISMISSED** [review]
I verified the Chain IDs using [asciitohex.com](https://www.asciitohex.com) and they look correct. The rest of the changes seem to match the pattern of other chains to me.

File: shared/constants/network.ts (hunk 1 above)
Comment (DenisCarriere) [reply shared/constants/network.ts]: Updated ✅  to "EOS EVM"

**Review by kj4ezj — APPROVED** [review]
:shipit:

File: shared/constants/network.ts (hunk 1 above)
Comment (kj4ezj) [reply shared/constants/network.ts]: Thank you! I don't have permissions to mark this thread "resolved" I guess, but...resolved!

**Pull request closed without merging**